File: python/observer/NewsLetter/subject/Newsletter.py
from observer.Observer import Observer
from subject.Subject import Subject

class Newsletter(Subject):

    def __init__(self):
        self.__observers = []
        self.__name = ''
        self.__observer = Observer()

    def register_observer(self, new_observer):
        self.__observers.append(new_observer)

    def remove_observer(self, remove_observer):
        index = self.__observers.index(remove_observer)
        self.__observers.pop(index)

    def notify_observers(self):
        for self.__observer in self.__observers:
            self.__observer.update(self.__name)

    def set_name(self, name):
        self.__name = name
        self.notify_observers()

File: C++/Observer/NewsLetter/subject/Newsletter.cpp

#include "Subject.h"
#include <utility>
#include <vector>

using namespace std;

class Newsletter : Subject {

public:

    void registerObserver(Observer *newObserver) override {
        observers.push_back(newObserver);
    }

    void removeObserver(Observer *removeObserver) override {
        observers.erase(
                remove(observers.begin(), observers.end(), removeObserver),
                observers.end()
        );
    }

    void notifyObservers() override {
        for (auto & observer : observers) {
            observer->update(_name);
        }
    }

    void setName(string name) {
        _name = move(name);
        notifyObservers();
    }

private:
    vector<Observer*> observers;
    string _name;
};
